core/includes/database/mysql/database.inc

```diff
@@ -68,6 +68,11 @@ class DatabaseConnection_mysql extends DatabaseConnection {
       PDO::MYSQL_ATTR_USE_BUFFERED_QUERY => TRUE,
       // Because MySQL's prepared statements skip the query cache, because it's dumb.
       PDO::ATTR_EMULATE_PREPARES => TRUE,
+      // Convert numeric values to strings when fetching. In PHP 8.1,
+      // PDO::ATTR_EMULATE_PREPARES now behaves the same way as non emulated
+      // prepares and returns integers. See https://externals.io/message/113294
+      // for further discussion.
+      \PDO::ATTR_STRINGIFY_FETCHES => TRUE,
```

Comment:
I wasn't sure about including this change, since it may be hard to go back from, but my impression is that all integers are returned as strings currently, and this maintains that same behavior in PHP 8.1+. Additionally both D7 and D9 use this same change, so it will be expected with modules being ported. This seems like it's the best choice even if we don't particularly like it.